### PR TITLE
Release plans 2023

### PR DESCRIPTION
**Pull request by stianst**

### CODE REVIEW

**Review by mposolda — APPROVED** [review]

File: blog/2022/releases.adoc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+=== Highlights
+
+* Preview of the new Store
+* New Admin Console is graduated to the default console, while the old Admin Console is deprecated
+
+=== End of life
+
+* Old Account Console will be removed
+* Text-based login flows and authenticators will be removed
+* Some OpenID Connect adapters will be removed, including:
```
Comment (pedroigor) [inline blog/2022/releases.adoc:39]: As discussed, perhaps mention this blog post https://www.keycloak.org/2022/02/adapter-deprecation.

**Review by pedroigor — APPROVED** [review]

File: blog/2022/releases.adoc [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+* Some SAML adapters will be removed, including:
+** JBoss AS 7 and EAP 6
+** Jetty 9.2 and 9.3
+** WildFly legacy
+
+== Keycloak 20 - September/October
+
+=== Highlights
+
+* New store is graduated to the new default store for PostgreSQL and CockroachDB. We will come back with more details on what happens with the old store and support for other database vendors, but rest assured we will give everyone plenty of heads up, and at the minimum the old store will be supported at least until the middle of 2023.
```
[the remark this answers is not in the document]
Comment (abstractj) [reply blog/2022/releases.adoc:54]: Plus, might worth to change "old store" to "old Keycloak storage".

File: blog/2022/releases.adoc [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+=== End of life
+
+* Ability to upload custom JavaScript providers through REST APIs will be removed
+
+== Keycloak 19 - June/July
+
+=== Highlights
+
+* Preview of the new Store
+* New Admin Console is graduated to the default console, while the old Admin Console is deprecated
```
Comment (ssilvert) [inline blog/2022/releases.adoc:33]:
```suggestion
* New Admin UI is graduated to the default console, while the old Admin UI is deprecated
```

File: blog/2022/releases.adoc [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+_Subject to change: this is a provisional plan, which may be altered throughout the year._
+
+== Keycloak 18 - March/April
+
+=== Highlights
+
+* Enhancements and polishing for the Quarkus distribution
+* Preview of the new Kubernetes Operator for the Quarkus distribution
+* Preview of the new Admin Console
```
Comment (ssilvert) [inline blog/2022/releases.adoc:20]:
```suggestion
* Preview of the new Admin UI
```

File: blog/2022/releases.adoc [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+== Keycloak 19 - June/July
+
+=== Highlights
+
+* Preview of the new Store
+* New Admin Console is graduated to the default console, while the old Admin Console is deprecated
+
+=== End of life
+
+* Old Account Console will be removed
```
Comment (ssilvert) [inline blog/2022/releases.adoc:37]:
```suggestion
* Old Account UI will be removed
```

**Review by ssilvert — CHANGES_REQUESTED** [review]

File: blog/2022/releases.adoc [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+=== End of life
+
+* WildFly distribution will be removed
+* Legacy Kubernetes Operator will be removed
+
+== Keycloak 21 - December/January
+
+=== Highlights
+
+* With the new Quarkus distribution, new Operator, and new Store we are aiming to shift our focus onto production deployments of Keycloak. Keycloak 21 is the release where we're aiming all this effort to really come together to make it as easy as possible to install and manage Keycloak at any scale.
```
Comment (abstractj) [inline blog/2022/releases.adoc:65]: 2 cents: "With the new releases of Quarkus distribution, Keycloak Operator and Keycloak Storage we are aiming..."

File: blog/2022/releases.adoc [hunk 7]
```diff
@@ -0,0 +1,81 @@
+:title: Keycloak release plans for 2023
+:date: 2022-03-22
+:publish: true
+:author: Stian Thorgersen
+
+December last year was a little bit on the crazy side with 3 feature releases of Keycloak (15.1, 16.0, and 16.1). This was down to balancing WildFly upgrades with introduction of the Quarkus dist preview.
+
+This year we are planning to bring more predictability to Keycloak releases and are aiming for a quarterly release, with more frequent patch releases in-between.
```
Comment (abstractj) [inline blog/2022/releases.adoc:8]: "This year, we are..."

File: blog/2022/releases.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (andymunro) [reply blog/2022/releases.adoc:20]: The name should be consistent and match what customers call it, which I believe is Admin Console.  The documentation calls it Admin Console, although sometimes it is called "admin console."  We are updating it gradually.

The new "UI" calls it "Administration Console" on the Welcome page and loading "admin console" when it starts.  I recommend  "Admin Console" across the board.  That would require the least change and match what customers call it.

File: blog/2022/releases.adoc (hunk 5 above)
Comment (andymunro) [reply blog/2022/releases.adoc:37]: How about:
Old Account Console, also called the User Account Service, will be removed, but the new Account Console will remain.

**Review by hmlnarik — APPROVED** [review]
Thanks!

**Review by andymunro — CHANGES_REQUESTED** [review]
A few suggestions on terminology, @stianst.

File: blog/2022/releases.adoc [hunk 8]
```diff
@@ -0,0 +1,81 @@
+:title: Keycloak release plans for 2023
+:date: 2022-03-22
+:publish: true
+:author: Stian Thorgersen
+
+December last year was a little bit on the crazy side with 3 feature releases of Keycloak (15.1, 16.0, and 16.1). This was down to balancing WildFly upgrades with introduction of the Quarkus dist preview.
```
[the remark this answers is not in the document]
Comment (stianst) [reply blog/2022/releases.adoc:6]:
```suggestion
December last year was a bit on the crazy side with 3 feature releases of Keycloak (15.1, 16.0, and 16.1). This was down to balancing WildFly upgrades with introduction of the Quarkus dist preview.
```

File: blog/2022/releases.adoc (hunk 7 above)
Comment (stianst) [reply blog/2022/releases.adoc:8]:
```suggestion
This year, we are planning to bring more predictability to Keycloak releases and are aiming for a quarterly release, with more frequent patch releases in-between.
```

File: blog/2022/releases.adoc [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+This year we are planning to bring more predictability to Keycloak releases and are aiming for a quarterly release, with more frequent patch releases in-between.
+
+One thing worth highlighting is we have decided to extend the support of the WildFly distribution until September to give everyone more time to migrate.
+
+_Subject to change: this is a provisional plan, which may be altered throughout the year._
```
[the remark this answers is not in the document]
Comment (stianst) [reply blog/2022/releases.adoc:12]:
```suggestion
_Subject to change: this is a provisional plan, which may change throughout the year._
```

File: blog/2022/releases.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (stianst) [reply blog/2022/releases.adoc:20]: @ssilvert I'm sticking with Admin Console for now as that's what the community currently knows it as. Also, had a quick look at https://github.com/keycloak/keycloak-admin-ui and it's referred to as Admin Console. Let's have a chat about if we want to actually go ahead with renaming it separately.

Comment (stianst) [reply blog/2022/releases.adoc:20]: See above

File: blog/2022/releases.adoc (hunk 5 above)
Comment (stianst) [reply blog/2022/releases.adoc:37]:
```suggestion
* Old Account Console will be removed, but the new Account Console will remain of course
```

File: blog/2022/releases.adoc [hunk 10]
```diff
@@ -0,0 +1,81 @@
+:title: Keycloak release plans for 2023
+:date: 2022-03-22
```
Comment (stianst) [inline blog/2022/releases.adoc:2]:
```suggestion
:date: 2022-03-24
```

File: blog/2022/releases.adoc (hunk 1 above)
Comment (stianst) [reply blog/2022/releases.adoc:39]:
```suggestion
* Some OpenID Connect adapters will be removed (https://www.keycloak.org/2022/02/adapter-deprecation[adapter deprecation blog post]), including:
```

File: blog/2022/releases.adoc [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+=== End of life
+
+* Old Admin Console will be removed
+
+== Keycloak 22 - March/April 2023
+
+=== End of life
+
+* Remaining deprecated OpenID Connect adapters will be removed, including:
```
Comment (stianst) [inline blog/2022/releases.adoc:75]:
```suggestion
* Remaining deprecated OpenID Connect adapters will be removed (https://www.keycloak.org/2022/02/adapter-deprecation[adapter deprecation blog post]), including:
```

File: blog/2022/releases.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (stianst) [reply blog/2022/releases.adoc:54]: We've been calling it old and new store up to now in loads of places, so going to stick with that

**Pull request merged**